samcli/lib/config/samconfig.py

```diff
@@ -183,6 +197,46 @@ def _write(self):
     def _version(self):
         return self.document.get(VERSION_KEY, None)
 
+    def _deduplicate_global_parameters(self, cmd_name_key, section, key, env=DEFAULT_ENV):
+        """
+        In case the global parameters contains the same key-value pair with command parameters,
+        we only keep the entry in global parameters
+
+        Parameters
+        ----------
+        cmd_name_key : str
+            key of command name
+
+        section : str
+            Specific section within the command to look into. e.g. `parameters`
+
+        key : str
+            Key to write the data under
+
+        env : str
+            Optional, Name of the environment
+        """
+        global_params = self.document.get(env, {}).get(DEFAULT_GLOBAL_CMDNAME, {}).get(section, {})
+        command_params = self.document.get(env, {}).get(cmd_name_key, {}).get(section, {})
+        if (
+            cmd_name_key != DEFAULT_GLOBAL_CMDNAME
+            and global_params
+            and command_params
+            and global_params.get(key)
+            and global_params.get(key) == command_params.get(key)
+        ):
+            value = command_params.get(key)
+            save_global_message = (
+                f'\n\tParameter "{key}={value}" in [{env}.{cmd_name_key}.{section}] is defined as a global '
+                f"parameter [{env}.{DEFAULT_GLOBAL_CMDNAME}.{section}].\n\tThis parameter will be only saved "
+                f"under [{env}.{DEFAULT_GLOBAL_CMDNAME}.{section}] in {self.filepath}."
+            )
+            LOG.info(save_global_message)
+            # Only keep the global parameter
+            items = self.document[env][cmd_name_key][section].copy()
+            items.pop(key)
+            self.document[env][cmd_name_key][section] = items
```

Comment:
Why the copy? If this is just deleting `key`, `del self.document[env][cmd_name_key][section][key]` should be enough.

Reply:
No special preference on these two options.

Reply:
Then `del` makes more sense as it's obvious what's going on and avoids unnecessary copy.

Reply:
updated with del.